src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,424 @@
+        dest_dir, file = dir_has_onefile
+        list_fd_bef_copy = os.listdir(dest_dir)
+        copy_tree(source_dir, dest_dir)
+        list_fd_after_copy = os.listdir(dest_dir)
+        assert len(list_fd_bef_copy) == len(list_fd_after_copy)
+
+    def test_dir_has_onefile(self, dir_has_onefile, empty_dir):
+        """
+        When the Source Directory has only one file
+        """
+        source_dir, source_file = dir_has_onefile
+        dest_dir = empty_dir
+        copy_tree(source_dir, dest_dir)
+        copied_file = os.path.join(dest_dir, os.path.basename(source_file))
+        assert os.path.exists(copied_file)
+
+    def test_dir_has_onedir(self, dir_has_onedir, empty_dir):
+        """
+        When the Source Directory has only one sub directory
+        """
+        source_dir, sub_dir = dir_has_onedir
+        dest_dir = empty_dir
+        copy_tree(source_dir, dest_dir)
+        copied_dir = os.path.join(dest_dir, os.path.basename(sub_dir))
+        assert os.path.exists(copied_dir)
+
+    def test_dir_has_nestedfd(self, empty_dir, nested_fd):
+        """
+        When the files are nested in subdirectories more than 2 directories deep
+        """
+        dest_dir = empty_dir
+        source_dir, sub_dir1, sub_dir2, source_file = nested_fd
+        copy_tree(source_dir, dest_dir)
+        copied_dir1 = os.path.join(dest_dir, os.path.basename(sub_dir1))
+        copied_dir2 = os.path.join(copied_dir1, os.path.basename(sub_dir2))
+        copied_file = os.path.join(copied_dir2, os.path.basename(source_file))
+        assert os.path.exists(copied_dir1)
+        assert os.path.exists(copied_dir2)
+        assert os.path.exists(copied_file)
+
+
+def test_ignore_missing_dir_error():
+    """
+    Checks whether the missing directory error is ignored
+    when we try to remove a directory which is not exist
+    """
+    dir = tempfile.mkdtemp()
+    shutil.rmtree(dir)
+    shutil.rmtree(dir, onerror=ignore_missing_dir_error)
+
+
+class TestMoveTree:
+    """
+    Checks that all the contents are moved from source to destination
+    """
+
+    def test_empty_dir(self, empty_dir, dir_has_onedir):
+        """
+        When the Source Directory is empty
+        """
+        source_dir = tempfile.mkdtemp()
+        dest_dir, sub_dir = dir_has_onedir
+        list_fd_bef_move = os.listdir(dest_dir)
+        move_tree(source_dir, dest_dir)
+        list_fd_after_move = os.listdir(dest_dir)
+        assert len(list_fd_bef_move) == len(list_fd_after_move)
+
+    def test_dir_has_onefile(self, empty_dir):
+        """
+        When the Source Directory has only one file
+        """
+        source_dir = tempfile.mkdtemp()
+        fd, source_file = tempfile.mkstemp(dir=source_dir)
+        dest_dir = empty_dir
+        move_tree(source_dir, dest_dir)
+        moved_file = os.path.join(dest_dir, os.path.basename(source_file))
+        assert os.path.exists(moved_file) and not os.path.exists(source_file)
```

Comment:
Break these up into separate assertions (throughout). It's not good to have multiple conditions tested by the same `assert`; if the line fails, you don't know which one failed.